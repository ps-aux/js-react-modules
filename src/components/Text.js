import React from 'react'

const Text = ({children, value, ...rest}) =>
    <span {...rest}>
        {children || value}
    </span>

export default Text
