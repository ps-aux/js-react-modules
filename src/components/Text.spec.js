import React from 'react'
import renderer from 'react-test-renderer'
import Text from './Text'


describe('Text renders', () => {
    it('children content properly', () => {
        const comp = renderer.create(
            <Text>Text</Text>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('text as value properly', () => {
        const comp = renderer.create(
            <Text value="Text"/>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('additional className', () => {
        const comp = renderer.create(
            <Text value="any" className="additional"/>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
