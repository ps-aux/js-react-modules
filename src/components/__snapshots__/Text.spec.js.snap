// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders additional className 1`] = `
<span
  className="text additional"
>
  any
</span>
`;

exports[`Text renders children content properly 1`] = `
<span
  className="text"
>
  Text
</span>
`;

exports[`Text renders text as value properly 1`] = `
<span
  className="text"
>
  Text
</span>
`;
